packages/editor-ui/src/views/WorkflowEvaluation/EvaluationEditView.vue

```diff
@@ -0,0 +1,243 @@
+import { useAnnotationTagsStore } from '@/stores/tags.store';
+import { useDebounce } from '@/composables/useDebounce';
+import { useEvaluationForm } from '@/components/WorkflowEvaluation/composables/useEvaluationForm';
+
+import EvaluationHeader from '@/components/WorkflowEvaluation/EditEvaluation/EvaluationHeader.vue';
+import DescriptionInput from '@/components/WorkflowEvaluation/EditEvaluation/DescriptionInput.vue';
+import EvaluationStep from '@/components/WorkflowEvaluation/EditEvaluation/EvaluationStep.vue';
+import TagsInput from '@/components/WorkflowEvaluation/EditEvaluation/TagsInput.vue';
+import WorkflowSelector from '@/components/WorkflowEvaluation/EditEvaluation/WorkflowSelector.vue';
+import MetricsInput from '@/components/WorkflowEvaluation/EditEvaluation/MetricsInput.vue';
+
+const props = defineProps<{
+	testId?: number;
+}>();
+
+const router = useRouter();
+const route = useRoute();
+const locale = useI18n();
+const { debounce } = useDebounce();
+const toast = useToast();
+const { isLoading, allTags, tagsById, fetchAll } = useAnnotationTagsStore();
+
+const testId = computed(() => props.testId ?? (route.params.testId as unknown as number));
+const buttonLabel = computed(() =>
+	testId.value
+		? locale.baseText('workflowEvaluation.edit.updateTest')
+		: locale.baseText('workflowEvaluation.edit.saveTest'),
+);
+
+const {
+	state,
+	fieldsIssues,
+	isSaving,
+	loadTestData,
+	saveTest,
+	startEditing,
+	saveChanges,
+	cancelEditing,
+	handleKeydown,
+} = useEvaluationForm();
+
+onMounted(async () => {
+	await fetchAll();
+	if (testId.value) {
+		await loadTestData(testId.value);
+	}
+});
+
+async function onSaveTest() {
+	try {
+		const newTest = await saveTest(testId.value);
+		if (newTest) {
+			await router.replace({
+				name: VIEWS.WORKFLOW_EVALUATION_EDIT,
+				params: { testId: newTest.id },
+			});
+		}
+		toast.showMessage({
+			title: locale.baseText('workflowEvaluation.edit.testSaved'),
+			type: 'success',
+		});
+	} catch (e: unknown) {
+		toast.showError(e, locale.baseText('workflowEvaluation.edit.testSaveFailed'));
+	}
+}
+
+function hasIssues(key: string) {
+	return fieldsIssues.value.some((issue) => issue.field === key);
+}
+
+watch(
+	() => state.value,
+	debounce(
+		async () => {
+			if (state.value.evaluationWorkflow.value) {
+				await onSaveTest();
+			}
+		},
+		{ debounceTime: 1000 },
+	),
+	{ deep: true },
+);
+</script>
+
+<template>
+	<div :class="$style.container">
+		<div :class="$style.content">
+			<EvaluationHeader
+				v-model="state.name"
+				:class="{ 'has-issues': hasIssues('name') }"
+				:start-editing="startEditing"
+				:save-changes="saveChanges"
+				:handle-keydown="handleKeydown"
+			/>
+
+			<DescriptionInput v-model="state.description" />
+
+			<div :class="$style.panelIntro">When running a test</div>
+			<BlockArrow :class="$style.introArrow" />
+			<div :class="$style.panelBlock">
+				<EvaluationStep :class="$style.step" :title="'Fetch 5 past executions'">
```

Comment:
Would it make sense to remove number of executions for now? Because fetching the real number might require some time, that might be spent on something more important

Reply:
Will adjust this in the next stage — mocking nodes and selecting executions